pyccl/tk3d.py

```diff
@@ -133,10 +133,39 @@ def __init__(self, a_arr, lk_arr, tkk_arr=None,
                                                         int(is_logt), status)
         check(status)
 
+    def __eq__(self, other):
+        # Check the object class.
+        if self.__class__ is not other.__class__:
+            return False
+        # If the objects contain no data, return early.
+        if not (self or other):
+            return True
+        # If one is factorizable and the other one is not, return early.
+        if self.tsp.is_product ^ other.tsp.is_product:
+            return False
+        # Check extrapolation orders.
+        if not (self.extrap_order_lok == other.extrap_order_lok
+                and self.extrap_order_hik == other.extrap_order_hik):
+            return False
+        # Check the individual splines.
+        a1, lk11, lk12, tk1 = self.get_spline_arrays()
+        a2, lk21, lk22, tk2 = other.get_spline_arrays()
+        return ((a1 == a2).all()
+                and (lk11 == lk21).all() and (lk21 == lk22).all()
+                and np.allclose(tk1, tk2, atol=0, rtol=1e-12))
```

Comment:
@damonge  same here as with `Pk2D`.

Reply:
Same answer

Reply:
See comment in pk2d about splitting this

Reply:
done re. tol set to zero

Reply:
array_equal. We don't need the extra complexity or the extra operations allclose performs

Reply:
done